api/src/modules/h3-data/services/h3-filter-years-by-layer.service.ts

```diff
@@ -89,4 +92,62 @@ export class H3FilterYearsByLayerService {
   ): Promise<number[]> {
     return this.sourcingRecordService.getYears(materialIds);
   }
+
+  async getH3MaterialYearForCalculations(
+    materialId: string,
+    materialType: MATERIAL_TO_H3_TYPE,
+    year: number,
+  ): Promise<number | undefined> {
+    let materialDataYear: number | undefined;
+    const availableH3DataYears: number[] =
+      await this.getAvailableYearsForH3MaterialData(materialId, materialType);
+
+    materialDataYear = availableH3DataYears.includes(year)
+      ? year
+      : availableH3DataYears.find((el: number) => el < year);
+
+    if (!materialDataYear)
+      materialDataYear = availableH3DataYears
+        .reverse()
+        .find((el: number) => el > year);
+
+    return materialDataYear;
+  }
+
+  async getH3IndicatorYearForCalculations(
```

Comment:
I have some suggestions here:

1. At a very personal level, I find the name a bit weird to understand what is it's purpose, as a person who has not been involved lately in this fallback strategy. Either a jsdocs documentation or maybe a more meaningful name (or both?) as `getClosestAvailableYearForMaterialH3` would be better. Thoughts?

2. In line 103 you are using  a method which is basically a wrapper for a method implemented in h3-data-repository, with no additional logic in it. Since this service receives the singleton instance of said repository, I think you can directly access said method avoiding adding 
 -  getAvailableYearsForH3MaterialData
 - getAvailableYearsForH3IndicatorData
 methods.
 
 Also, but this is out of this PR's scope, I think it would be a good idea to handle the error for when no year could be found inside these methods, which is currently happening outside.

Reply:
Thanks a lot for suggestions - all implemented.